File: app/src/main/java/com/mkrworld/libfilter/filter/overlayfilter/MultiplierEffect.java
package com.mkrworld.libfilter.filter.overlayfilter;

import com.mkrworld.libfilter.jnicaller.Effector;

public class MultiplierEffect extends BaseOverlayFilter {

    /**
     * Constructor
     *
     * @param pixelArray        Array of Pixel
     * @param imageWidth
     * @param overlayPixelArray
     * @param multiplier
     */
    public MultiplierEffect(int[] pixelArray, int imageWidth, int[] overlayPixelArray, float multiplier) {
        super(pixelArray, imageWidth, overlayPixelArray, multiplier);
    }

    @Override
    public int[] applyFilter() throws Exception {
        return Effector.setMultiplyEffect(getPixelArray(), getImageWidth(), getOverlayPixelArray(), getMultiplier());
    }
}

File: app/src/main/java/com/mkrworld/libfilter/filter/MultiplierEffect.kt
package com.mkrworld.libfilter.filter

import com.mkrworld.libfilter.jnicaller.Effector

/**
 * @author THEMKR
 */
class MultiplierEffect : BaseOverlayFilter {

    /**
     * Constructor
     *
     * @param pixelArray        Array of Pixel
     * @param imageWidth
     * @param overlayPixelArray
     * @param multiplier
     */
    constructor(pixelArray: IntArray, imageWidth: Int, overlayPixelArray: IntArray, multiplier: Float) : super(pixelArray, imageWidth, overlayPixelArray, multiplier)

    @Throws(Exception::class)
    override fun applyFilter(): IntArray {
        return Effector.setMultiplyEffect(pixelArray, imageWidth, overlayPixelArray, multiplier)
    }
}
